mne/preprocessing/artifact_detection.py

```diff
@@ -4,12 +4,99 @@
 
 
 import numpy as np
-
 from ..annotations import (Annotations, _annotations_starts_stops)
 from ..transforms import (quat_to_rot, _average_quats, _angle_between_quats,
                           apply_trans, _quat_to_affine)
+from ..filter import filter_data
 from .. import Transform
-from ..utils import (_mask_to_onsets_offsets, logger)
+from ..utils import (_mask_to_onsets_offsets, logger, verbose)
+
+
+@verbose
+def annotate_muscle_zscore(raw, threshold=4, picks=None, min_length_good=.1,
+                           filter_freq=(110, 140), n_jobs=1, verbose=None):
+    """Detect segments with muscle artifacts.
+
+    Detects segments periods that contain high frequency activity beyond the
+    specified threshold. Muscle artifacts are most detectable in the range of
+    110-140 Hz.
+
+    Raw data is band-pass filtered between ``filter_freq`` especified
+    frequencies (default is 110 - 140 Hz), the signal envelope computed,
+    z-scored across samples, channel summation and division by the square root
+    of the channel number, and low-pass filtered to better capture beginning
+    and end of muscle activity and false positive transient peaks.
```

Comment:
There was some discussion here before, the conclusion was that it is more a heuristic... 
Channel types have different sensitivity, better just use one type.

Reply:
Quoting @bloyl message in the PR:
> > do we reinvent the wheel here? Can you provide more academic refs?
> 
> I don't know if approaches like this have been published, although I assume that they have. And I don't know how this approach compares with the methods that the google search turned up.
> 
> I use a method like this, to replace a time consuming manual check of the data. The tweaks, low pass filtering the envelope, filling in short artifact free periods, extending onsets/offsets etc. were added to better replicate the manual process that our clinical technologists were doing. I have not done a comparative study or published a paper on it because while useful its not that interesting.
> 
> At the end of the day, this is heuristic z-score outlier detection that is able to mark periods of time that if you manually reviewed the data you would probably want to remove.

I didn't find a paper on z-score muscle detection. 

Is information ever too much? I thought it would be useful for a user, especially without much python knowledge, to know what the function does without having to dive into the code.

Reply:
Please look back at my original suggestion, which was "for background information affecting choice of filter and threshold parameters, see XXX and YYY".  I was *not* asking for a paper that justifies the specific computations that we are doing here.

@bloyl made a suggestion [here](https://github.com/mne-tools/mne-python/pull/7407#discussion_r393657015) and @agramfort made a suggestion [here](https://github.com/mne-tools/mne-python/pull/7407#pullrequestreview-375828439) about related papers. Have you looked at those?  Are they appropriate / relevant with regard to *what frequencies are present in muscle artifacts* (relevant to bandpass freqs) and/or *the amplitude of artifacts relative to brain signals* (relevant to threshold)?

> Is information ever too much?

Well for one thing, we have limits on how long docstrings are allowed to be (800 characters I think). I'm 50/50 on whether the exact details of the computations are important here. If you feel strongly that they should stay, then here's a suggestion for the docstring that I think clarifies things:

> Create annotations for segments that likely contain muscle artifacts.
>
> Detects data segments containing activity in the frequency range given by
> ``filter_freq`` whose envelope magnitude exceeds the specified z-score
> threshold (when summed across channels and divided by ``sqrt(n_channels)``).
> False-positive transient peaks are prevented by low-pass filtering the
> resulting z-score time series at 4 Hz. Only operates on a single channel
> type, so ``pick_types`` or similar may be required before using. See REF1,
> REF2 for background on choosing ``filter_freq`` and ``threshold``.

Reply:
Nice docstring @drammock 
Yes, I saw those papers, they mention the freq. range of the muscle activity, but that might not be the ideal ``filter_freq`` range for this function. One says that ocular muscle artifacts peak at around 30-40Hz, but for example, in the example plot of this PR, the annotated muscle artifact is likely from an eye blinking (it overlaps with a blink). In my experience, 110-140Hz works very well in MEG, a lab colleague was also using it for EEG. Maybe this high freq. band it is easier to detect muscle activity because is more silent from sensor noise (or brain activity).
The ``threshold`` is dependent on the sensor sensitivity and scalp distribution.

If you still think that a reference would be good, let me know and I will look at how to put this [one](https://github.com/mne-tools/mne-python/pull/7407#discussion_r393657015)

Reply:
If there's a default like `(110, 140)`, people are going to wonder "why those values?" and "is it OK to change them?" and "how much is it OK to change them?"  The docstring should answer those questions, or at least it should say "see this example / tutorial / published paper, for more info".  If there are no published papers that say "110-140 Hz is best", then we should point to a paper that says "here are the physiologically plausible / relevant frequencies to consider", so that users at least know that there is no point trying, say, 8 Hz or 800 Hz. So yes, I still think that a reference would be good. If a more extended discussion is needed, it could go in a `Notes` section of the docstring rather than up front.

You can look at the source of `tutorials/preprocessing/plot_60_maxwell_filtering_sss.py` to see how we do references with `footcite` and `footbibliography`, and `doc/references.bib` for the BibTeX source.